File: java/inheritance/Inheritance.java
class Base {
    int x;

    Base(int a) {
        x = a;
        System.out.println("Base Class Constructor: x = " + x);
    }
}

class Derived extends Base {
    int y;

    Derived(int a, int b) {
        super(a);
        y = b;
        System.out.println("Derived Class Constructor: x = " + x + " y = " + y);
    }
}

public class Inheritance {
    public static void main(String args[]) {
        Derived ob = new Derived(10, 20);
    }
}

File: cpp/inheritance/inheritance.cpp
#include <iostream>
using namespace std;

class Base
{
public:
    int x;

    Base(int a) : x(a) // Constructor alt. syntax
    {
        cout << "Base Class Constructor: x = " << x << endl;
    }

    ~Base() // Destructor
    {
        cout << "Destructor of Base Class" << endl;
    }
};

class Derived : public Base
{
public:
    int y;

    Derived(int a, int b) : Base(a)
    {
        y = b;
        cout << "Derived Class Constructor: x = " << x << " y = " << y << endl;
    }

    ~Derived()
    {
        cout << "Destructor of Derived Class" << endl;
    }
};

int main()
{
    Derived d(10, 20);
    return 0;
}